File: Back/ClinicalTemplateApi/AppInfrastructure/Entities/Appointment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AppInfrastructure.Contracts
{
    public class Appointment
    {
        [Key]
        public int Id { get; set; }
        public AppointmentType Type { get; set; }
        public Patient Patient { get; set; }
        public int PatientId { get; set; }

        [DataType(DataType.Date)]
        [Column(TypeName = "Date")]
        public DateTime Date { get; set; }

        [DataType(DataType.Time)]
        public TimeSpan Hour { get; set; }

    }
}

File: Front/clinicalTemplate/src/app/auth/models/appointment.ts
import { Time } from '@angular/common';

export class Appointment {
    Id: string;
    Type: AppointmentType;
    PatientId: number;
    Date: Date;
    Hour: Time;
}

export enum AppointmentType {
        GeneralMedicine = 1,
        Odontology = 2,
        Pediatrics = 3,
        Neurology = 4
}
